Tests/PowerSkillTests/CustomEntitySearchTests/CustomEntitySearchHelpers.cs

```diff
@@ -76,14 +85,60 @@ public static string BuildOutput(string[] entities, string[] matches, int[] matc
                 expectedOutput = expectedOutput.Replace(@"""errors"":[]", errorMessage);
                 expectedOutput = expectedOutput.Remove(35, 32);
             }
-            Assert.AreEqual(expectedOutput, actualOutput);
+
+            Helpers.AssertJsonEquals(expectedOutput, actualOutput);
         }
 
         public static async Task CallEntitySearchFunctionAndCheckResults(
-            string[] expectedFoundEntities, string[] expectedMatches, int[] expectedMatchIndices,
-            string text, string[] words, string warningMessage = "", string errorMessage = "")
+            string text,
+            string[] words,
+            string expectedOutput,
+            Dictionary<string, string[]> synonyms = null,
+            string[] exactMatches = null,
+            int offset = 0)
         {
-            string input = BuildInput(text, words);
+            string input = BuildInput(text);
+            ReplaceWordsJson(words, synonyms, exactMatches, offset);
+            string actualOutput = await QueryEntitySearchFunctionAndSerialize(input);
+
+            Helpers.AssertJsonEquals(expectedOutput, actualOutput);
+        }
+
+        public static void ReplaceWordsJson(
```

Comment:
Can you rename this to make it more explicit it's replacing the words file? Like maybe `ReplaceWordsJsonFile` or `UpdateWordsJsonFile`?